### PR TITLE
Remove client tags and add more tests for parsing

### LINKED ISSUE #230: Increase test coverage for logic

### LINKED ISSUE #248: Remove "tags" field from Client

### PR DESCRIPTION
**Pull request by tankangliang**
## Description

Removed `tags` field from `Client`

Fixes #248 
Related: #230 

## Testing

All tests are passing as usual.

Updated test to reflect changes to country commands and a little bit of standardization in format.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ClientEditCommand.java [hunk 1]
```diff
@@ -163,15 +154,14 @@ public EditClientDescriptor(EditClientDescriptor toCopy) {
...
-            setTags(toCopy.tags);
         }
 
         /**
          * Returns true if at least one field is edited.
          */
         public boolean isAnyFieldEdited() {
             return CollectionUtil.isAnyNonNull(name, phone, email, address, country, timezone,
-                    contractExpiryDate, tags);
+                    contractExpiryDate);
```
Comment (qwoprocks) [inline src/main/java/seedu/address/logic/commands/ClientEditCommand.java]: can this be put on the same line

File: src/main/java/seedu/address/logic/parser/ClientAddCommandParser.java [hunk 2]
```diff
@@ -61,10 +58,9 @@ public ClientAddCommand parse(String args) throws ParseException {
             contractExpiryDate = ContractExpiryDate.NULL_DATE;
         }
         LastModifiedInstant lastModifiedInstant = new LastModifiedInstant();
-        Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
 
         Client client = new Client(name, phone, email, address, country, timezone, contractExpiryDate,
-                lastModifiedInstant, tagList);
+                lastModifiedInstant);
```
Comment (qwoprocks) [inline src/main/java/seedu/address/logic/parser/ClientAddCommandParser.java:63]: same line?

File: src/test/java/seedu/address/testutil/ClientBuilder.java [hunk 3]
```diff
@@ -162,7 +146,7 @@ public ClientBuilder withLastModifiedInstant(String lastModifiedInstant) {
      */
     public Client build() {
         return new Client(name, phone, email, address, country, timezone, contractExpiryDate,
-                lastModifiedInstant, tags);
+                lastModifiedInstant);
```
Comment (qwoprocks) [inline src/test/java/seedu/address/testutil/ClientBuilder.java]: same line?

File: src/test/java/seedu/address/model/client/ClientTest.java [hunk 4]
```diff
@@ -46,16 +38,16 @@ public void isSameClient() {
...
         // same name, same email, different attributes -> returns true
         editedAlice = new ClientBuilder(ALICE).withPhone(VALID_PHONE_BOB).withAddress(VALID_ADDRESS_BOB)
-                .withCountry(VALID_COUNTRY_BOB).withTags(VALID_TAG_HUSBAND).build();
+                .withCountry(VALID_COUNTRY_BOB).build();
         assertTrue(ALICE.isSameClient(editedAlice));
 
         // same name, same phone, same email, different attributes -> returns true
-        editedAlice = new ClientBuilder(ALICE).withAddress(VALID_ADDRESS_BOB).withTags(VALID_TAG_HUSBAND)
+        editedAlice = new ClientBuilder(ALICE).withAddress(VALID_ADDRESS_BOB)
                 .withCountry(VALID_COUNTRY_BOB).withLastModifiedInstant(VALID_LAST_MODIFIED_INSTANT_BOB).build();
```
Comment (qwoprocks) [inline src/test/java/seedu/address/model/client/ClientTest.java]: probably can reformat here abit

File: src/test/java/seedu/address/logic/parser/ClientEditCommandParserTest.java [hunk 5]
```diff
@@ -187,23 +170,17 @@ public void parse_oneFieldSpecified_success() {
...
-                + TAG_DESC_FRIEND + PHONE_DESC_AMY + ADDRESS_DESC_AMY + EMAIL_DESC_AMY + TAG_DESC_FRIEND
-                + PHONE_DESC_BOB + ADDRESS_DESC_BOB + EMAIL_DESC_BOB + TAG_DESC_HUSBAND + COUNTRY_DESC_AMY
-                + COUNTRY_DESC_BOB + TIMEZONE_DESC_AMY + TIMEZONE_DESC_BOB;
+                + PHONE_DESC_AMY + ADDRESS_DESC_AMY + EMAIL_DESC_AMY + PHONE_DESC_BOB + ADDRESS_DESC_BOB
+                + EMAIL_DESC_BOB + COUNTRY_DESC_AMY + COUNTRY_DESC_BOB + TIMEZONE_DESC_AMY + TIMEZONE_DESC_BOB;
 
         EditClientDescriptor descriptor = new EditClientDescriptorBuilder().withPhone(VALID_PHONE_BOB)
-                .withEmail(VALID_EMAIL_BOB).withAddress(VALID_ADDRESS_BOB).withTags(VALID_TAG_FRIEND, VALID_TAG_HUSBAND)
+                .withEmail(VALID_EMAIL_BOB).withAddress(VALID_ADDRESS_BOB)
                 .withCountry(VALID_COUNTRY_BOB).withTimezone(VALID_TIMEZONE_BOB).build();
```
Comment (qwoprocks) [inline src/test/java/seedu/address/logic/parser/ClientEditCommandParserTest.java]: reformat

File: src/test/java/seedu/address/logic/parser/ClientEditCommandParserTest.java [hunk 6]
```diff
@@ -124,13 +108,12 @@ public void parse_invalidValue_failure() {
     @Test
     public void parse_allFieldsSpecified_success() {
         Index targetIndex = INDEX_SECOND_CLIENT;
-        String userInput = targetIndex.getOneBased() + PHONE_DESC_BOB + TAG_DESC_HUSBAND + COUNTRY_DESC_AMY
-                + EMAIL_DESC_AMY + ADDRESS_DESC_AMY + NAME_DESC_AMY + TAG_DESC_FRIEND + TIMEZONE_DESC_AMY;
+        String userInput = targetIndex.getOneBased() + PHONE_DESC_BOB + COUNTRY_DESC_AMY
+                + EMAIL_DESC_AMY + ADDRESS_DESC_AMY + NAME_DESC_AMY + TIMEZONE_DESC_AMY;
```
Comment (qwoprocks) [inline src/test/java/seedu/address/logic/parser/ClientEditCommandParserTest.java]: reformat

File: src/main/java/seedu/address/storage/JsonAdaptedClient.java [hunk 7]
```diff
@@ -33,28 +32,25 @@
...
-                @JsonProperty("contractExpiryDate") String contractExpiryDate,
-                @JsonProperty("lastModifiedInstant") String lastModifiedInstant,
-                @JsonProperty("tagged") List<JsonAdaptedTag> tagged,
-                @JsonProperty("clientNotes") List<JsonAdaptedNote> clientNotes) {
+                             @JsonProperty("phone") String phone, @JsonProperty("email") String email,
+                             @JsonProperty("address") String address, @JsonProperty("country") String country,
+                             @JsonProperty("timezone") String timezone,
+                             @JsonProperty("contractExpiryDate") String contractExpiryDate,
+                             @JsonProperty("lastModifiedInstant") String lastModifiedInstant,
+                             @JsonProperty("clientNotes") List<JsonAdaptedNote> clientNotes) {
```
Comment (qwoprocks) [inline src/main/java/seedu/address/storage/JsonAdaptedClient.java]: change back the indentation pls, supposed to be 8 spaces after the primary line, not 16

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 8]
```diff
@@ -29,42 +26,39 @@
...
-                getTagSet("colleagues"))
+                new ContractExpiryDate("9-6-2024"), new LastModifiedInstant("2020-06-06T00:00:00.000000Z"))
         };
     }
 
     public static ArrayList<CountryNote> getSampleCountryNotes() {
         return new ArrayList<>(Arrays.asList(
             new CountryNote("This is a country note", new Country("SG")),
-            new CountryNote("This is anotehr country note", new Country("MY"))
+            new CountryNote("This is another country note", new Country("MY"))
```
Comment (qwoprocks) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:54]: lmao good catch

**Review by qwoprocks — APPROVED** [review]
minor issues, other than that LGTM, also there's a tagged field left in duplicateClientAddressBook.json

File: src/main/java/seedu/address/logic/parser/ClientAddCommandParser.java (hunk 2 above)
Comment (tankangliang) [reply src/main/java/seedu/address/logic/parser/ClientAddCommandParser.java:63]: this goes past character limit

File: src/main/java/seedu/address/logic/commands/ClientEditCommand.java (hunk 1 above)
Comment (tankangliang) [reply src/main/java/seedu/address/logic/commands/ClientEditCommand.java]: done

File: src/test/java/seedu/address/testutil/ClientBuilder.java (hunk 3 above)
Comment (tankangliang) [reply src/test/java/seedu/address/testutil/ClientBuilder.java]: done

File: src/test/java/seedu/address/model/client/ClientTest.java (hunk 4 above)
Comment (tankangliang) [reply src/test/java/seedu/address/model/client/ClientTest.java]: first 2 are over char limit, last one reformatted

File: src/test/java/seedu/address/logic/parser/ClientEditCommandParserTest.java (hunk 5 above)
Comment (tankangliang) [reply src/test/java/seedu/address/logic/parser/ClientEditCommandParserTest.java]: done

File: src/test/java/seedu/address/logic/parser/ClientEditCommandParserTest.java (hunk 6 above)
Comment (tankangliang) [reply src/test/java/seedu/address/logic/parser/ClientEditCommandParserTest.java]: done

File: src/main/java/seedu/address/storage/JsonAdaptedClient.java (hunk 7 above)
Comment (tankangliang) [reply src/main/java/seedu/address/storage/JsonAdaptedClient.java]: done

**Pull request merged**